src/components/SearchBox/SearchBox.tsx

```diff
@@ -37,7 +37,7 @@ export type Props = {
    * The value of the search input when the state is externally controlled.
    */
   value?: string;
-} & HTMLProps<HTMLFormElement>;
+} & Omit<HTMLProps<HTMLFormElement>, "onChange" | "onSubmit" | "value">;
```

Comment:
We might want to revisit why we're explicitly defining `onChange`, `onSubmit` and `value` in the props type definition but for now at least there isn't an irreconcilable union type